src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,70 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["query", "msg", "say"];
+
+function parseInput(cmd, args, chan) {
+	switch (cmd) {
+	case "msg":
+	case "query":
+		return args.shift();
+	default:
+		return chan.name;
+	}
+}
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = parseInput(cmd, args, chan);
+
+	if (cmd === "query") {
+		if (!target) {
+			chan.pushMessage(this, new Msg({
+				type: Msg.Type.ERROR,
+				text: "You cannot open a query window without an argument.",
+			}));
+			return;
+		}
+
+		const query = network.getChannel(target);
```

Comment:
Hmmm, maybe we should have `targetName` instead of target and `target` instead of query?

Reply:
Since `query` is not used anywhere besides the `if` check, you could name it `existingChannel` to avoid even more confusion.

Or just move the `getChannel` call directly into the `if`.

Reply:
Oh yeah of course, because it's not actually the target we are trying to get. Yeah, I think moving the call into the if would be more clear. `if(network.getChannel(targetName))` is pretty obvious what it's trying to check.